server/src/main/java/io/crate/planner/node/ddl/CreateRolePlan.java

```diff
@@ -53,19 +69,77 @@ public StatementType type() {
     public void executeOrFail(DependencyCarrier dependencies,
                               PlannerContext plannerContext,
                               RowConsumer consumer,
-                              Row params, SubQueryResults subQueryResults) throws Exception {
-        SecureHash newPassword = UserActions.generateSecureHash(
+                              Row params,
+                              SubQueryResults subQueryResults) throws Exception {
+
+        Map<String, Object> properties = validate(
             createRole.properties(),
-            params,
             plannerContext.transactionContext(),
-            plannerContext.nodeContext());
+            plannerContext.nodeContext(),
+            params,
+            subQueryResults
+        );
+        SecureHash newPassword = UserActions.generateSecureHash(properties);
 
         if (createRole.isUser() == false && newPassword != null) {
             throw new UnsupportedOperationException("Creating a ROLE with a password is not allowed, " +
                                                     "use CREATE USER instead");
         }
 
-        roleManager.createRole(createRole.roleName(), createRole.isUser(), newPassword)
+        // Map is always not null to simplify streaming.
+        // It's safe to create a JwtProperties object as user input already passed validation.
+        Map<String, Object> jwtPropertiesMap = Maps.getOrDefault(properties, JWT_PROPERTY_KEY, Map.of());
+        JwtProperties jwtProperties = null;
+        if (jwtPropertiesMap.isEmpty() == false) {
+            String iss = Maps.get(jwtPropertiesMap, "iss");
+            String username = Maps.get(jwtPropertiesMap, "username");
+            jwtProperties = new JwtProperties(iss, username);
+        }
+
+        roleManager.createRole(createRole.roleName(), createRole.isUser(), newPassword, jwtProperties)
             .whenComplete(new OneRowActionListener<>(consumer, rCount -> new Row1(rCount == null ? -1 : rCount)));
     }
+
+    public static Map<String, Object> validate(GenericProperties<Symbol> genericProperties,
```

Comment:
I think if this is called validate it should only validate, not also convert the properties to values.
Could either split it, or rename to `parse` or `toValues` or something.

Reply:
renamed to parser